### ISSUE TITLE
OData query with non mapped fields

### ISSUE THREAD

**Issue by colradec** [dotnet/efcore#25433]
Hey guys,

I ran into some problems with queries generated for odata parameters.

When I am going to  ```$select=Date``` when fetching the WeatherForecast, the query succeeds normally with just the date field in the response, however when I am selecting ```$select=Summary``` field which is not mapped (V1WeatherForecast) but in the WeatherForecast class, the query ends in ```NullReferenceException: Object reference not set to an instance of an object.```.  This is caused by the ```ICollection``` but I am not sure if this is a bug from ef core/odata or the default behavior. 

###fix
When removing the relation to the Comments, the controller returns a default value ```null```, which is perfect.
This error can be fixed by adding ```.ToList().AsQueryable()``` when returning the IQueryable. So any ideas if this behavior is normal?

```C#
public class WeatherForecast
{
    [Key]
    public DateTime Date { get; set; }

    public int TemperatureC { get; set; }

    public int TemperatureF { get; set; }

    public string Summary { get; set; }

    public virtual ICollection<Comments> comments { get; set; }
}
```
```C#
public class V1WeatherForecast
{
    public DateTime Date { get; set; }

    public int TemperatureC { get; set; }

    public int TemperatureF { get; set; }

    public string Summary { get; set; }
    
    public IEnumerable<V1Comments> comments { get; set; }
}
```

```C#
return forecasts.Select(src => new V1WeatherForecast()
{
    Date = src.Date,
    TemperatureC = src.TemperatureC,
    TemperatureF = src.TemperatureF,
    comments = src.comments.Select(sel => new V1Comments()
    {
        comment = sel.comment,
        Date = sel.Date
    })
});
```

###the stack trace

```

    Microsoft.EntityFrameworkCore.Query.SqlExpressions.SelectExpression.ApplyCollectionJoin(int collectionIndex, int collectionId, Expression innerShaper, INavigationBase navigation, Type elementType, bool splitQuery)
    Microsoft.EntityFrameworkCore.Query.Internal.CollectionJoinApplyingExpressionVisitor.VisitExtension(Expression extensionExpression)
    System.Linq.Expressions.Expression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberAssignment(MemberAssignment node)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberBinding(MemberBinding node)
    System.Linq.Expressions.ExpressionVisitor.Visit<T>(ReadOnlyCollection<T> nodes, Func<T, T> elementVisitor)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberInit(MemberInitExpression node)
    System.Linq.Expressions.MemberInitExpression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    System.Linq.Expressions.ExpressionVisitor.VisitMember(MemberExpression node)
    System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    System.Linq.Expressions.ExpressionVisitor.VisitConditional(ConditionalExpression node)
    System.Linq.Expressions.ConditionalExpression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberAssignment(MemberAssignment node)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberBinding(MemberBinding node)
    System.Linq.Expressions.ExpressionVisitor.Visit<T>(ReadOnlyCollection<T> nodes, Func<T, T> elementVisitor)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberInit(MemberInitExpression node)
    System.Linq.Expressions.MemberInitExpression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberAssignment(MemberAssignment node)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberBinding(MemberBinding node)
    System.Linq.Expressions.ExpressionVisitor.Visit<T>(ReadOnlyCollection<T> nodes, Func<T, T> elementVisitor)
    System.Linq.Expressions.ExpressionVisitor.VisitMemberInit(MemberInitExpression node)
    System.Linq.Expressions.MemberInitExpression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    Microsoft.EntityFrameworkCore.Query.Internal.CollectionJoinApplyingExpressionVisitor.VisitExtension(Expression extensionExpression)
    System.Linq.Expressions.Expression.Accept(ExpressionVisitor visitor)
    System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
    Microsoft.EntityFrameworkCore.Query.RelationalQueryTranslationPostprocessor.Process(Expression query)
    Microsoft.EntityFrameworkCore.Query.QueryCompilationContext.CreateQueryExecutor<TResult>(Expression query)
    Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery<TResult>(Expression query, bool async)
    Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore<TResult>(IDatabase database, Expression query, IModel model, bool async)
    Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler+<>c__DisplayClass12_0<TResult>.<ExecuteAsync>b__0()
    Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery<TResult>(object cacheKey, Func<Func<QueryContext, TResult>> compiler)
    Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteAsync<TResult>(Expression query, CancellationToken cancellationToken)
    Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.ExecuteAsync<TResult>(Expression expression, CancellationToken cancellationToken)
    Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable<TResult>.GetAsyncEnumerator(CancellationToken cancellationToken)
    Microsoft.AspNetCore.Mvc.Infrastructure.AsyncEnumerableReader.ReadInternal<T>(object value)
    Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor.ExecuteAsyncEnumerable(ActionContext context, ObjectResult result, object asyncEnumerable, Func<object, Task<ICollection>> reader)
    Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0<TFilter, TFilterAsync>(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
    Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
    Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext<TFilter, TFilterAsync>(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
    Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
    Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
    Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
    Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
    Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
    Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
    Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
    Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

```
###Details

EF Core version: 5.08
Database provider: Postgres
Target framework: .NET 5.0
Operating system: Linux

**Comment by ajcvickers** [comment]
We believe this issue has been fixed in EF Core 6.0. Please try with the [latest preview](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/6.0.0-preview.7.21378.4) and get back to us if you are still hitting this issue.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*
